packages/babel-parser/test/fixtures/typescript/assert-predicate/asserts-this-with-predicate/output.json

```diff
@@ -0,0 +1,263 @@
+{
+  "type": "File",
+  "start": 0,
+  "end": 40,
+  "loc": {
+    "start": {
+      "line": 1,
+      "column": 0
+    },
+    "end": {
+      "line": 3,
+      "column": 1
+    }
+  },
+  "program": {
+    "type": "Program",
+    "start": 0,
+    "end": 40,
+    "loc": {
+      "start": {
+        "line": 1,
+        "column": 0
+      },
+      "end": {
+        "line": 3,
+        "column": 1
+      }
+    },
+    "sourceType": "module",
+    "interpreter": null,
+    "body": [
+      {
+        "type": "ClassDeclaration",
+        "start": 0,
+        "end": 40,
+        "loc": {
+          "start": {
+            "line": 1,
+            "column": 0
+          },
+          "end": {
+            "line": 3,
+            "column": 1
+          }
+        },
+        "id": {
+          "type": "Identifier",
+          "start": 6,
+          "end": 7,
+          "loc": {
+            "start": {
+              "line": 1,
+              "column": 6
+            },
+            "end": {
+              "line": 1,
+              "column": 7
+            },
+            "identifierName": "C"
+          },
+          "name": "C"
+        },
+        "superClass": null,
+        "body": {
+          "type": "ClassBody",
+          "start": 8,
+          "end": 40,
+          "loc": {
+            "start": {
+              "line": 1,
+              "column": 8
+            },
+            "end": {
+              "line": 3,
+              "column": 1
+            }
+          },
+          "body": [
+            {
+              "type": "TSDeclareMethod",
```

Comment:
This seems to be wrong

Reply:
Yeah it is wrong, it should be `ClassMethod`, in this [example](https://github.com/babel/babel/blob/0287c0f02f8844adbfd7d84a97a1a8a94c17fa1a/packages/babel-parser/test/fixtures/typescript/class/method-computed/input.ts#L2) the class property is also [finished](https://github.com/babel/babel/blob/0287c0f02f8844adbfd7d84a97a1a8a94c17fa1a/packages/babel-parser/test/fixtures/typescript/class/method-computed/output.json#L80) as TSDeclareMethod. I would suggest we address it in a separate PR.

Reply:
Well, in that example it's correct because there isn't the class body.

Reply:
Addressed in e6bf63e.